src/Features/CSharp/Portable/ImplementInterface/CSharpImplementInterfaceService.cs

```diff
@@ -120,28 +108,32 @@ protected override Document ImplementDisposePattern(Document document, SyntaxNod
     //   Dispose(false);
     // }}
 
-    // {CSharpFeaturesResources.This_code_added_to_correctly_implement_the_disposable_pattern}
     {(explicitly ? "void System.IDisposable." : "public void ")}Dispose()
     {{
         // {CSharpFeaturesResources.Do_not_change_this_code_Put_cleanup_code_in_Dispose_bool_disposing_above}
-        Dispose(true);
+        Dispose(true);";
+
+            if (classSymbol.IsSealed)
+            {
+                code += @$"
         // {CSharpFeaturesResources.TODO_colon_uncomment_the_following_line_if_the_finalizer_is_overridden_above}
         // GC.SuppressFinalize(this);
     }}
-    #endregion
 ";
+            }
+            else
+            {
+                code += @$"
+        GC.SuppressFinalize(this);
```

Comment:
📝 This should be using a semantically-correct reference to `System.GC` (using syntax generator APIs)

Reply:
good point.  will try to do that.  note: this code is nasty, it's really using direct code injection.  i'm going to see if i can get it off of that entirely.